mkosi/__init__.py

```diff
@@ -2102,7 +2102,7 @@ def invoke_dnf(
     if args.with_network == "never":
         cmdline += ["-C"]
 
-    if args.architecture is not None:
+    if args.architecture != args.host_architecture:
```

Comment:
While I know that calling `debootstrap` with `--arch` is not problematic, I was unsure about dnf's `--forcearch` and went for this check here.

It would be great if the dnf experts could tell me whether this makes sense.

Reply:
It's probably fine if CI passes